libevmasm/Assembly.cpp

```diff
@@ -74,6 +76,218 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
 	}
 }
 
+void Assembly::importAssemblyItemsFromJSON(Json::Value const& _code)
+{
+	solAssert(m_items.empty());
+	solRequire(_code.isArray(), AssemblyImportException, "Supplied JSON is not an array.");
+	for (Json::Value const& jsonItem: _code)
+		m_items.emplace_back(createAssemblyItemFromJSON(jsonItem, m_sourceList));
+
+	for (auto current = m_items.begin(); current != m_items.end(); ++current)
+	{
+		// Tags are very similar to `JUMPDEST`s. However, during the
+		// assembly json export a `JUMPDEST` is always generated after a `tag`.
+		// So we need to ignore exactly these `JUMPDEST`s during the import.
+		auto const nextIter = next(current);
+		if (
+			nextIter != m_items.end() &&
+			current->type() == AssemblyItemType::Tag &&
+			nextIter->type() == AssemblyItemType::Operation &&
+			nextIter->instruction() == Instruction::JUMPDEST
+		)
+			m_items.erase(nextIter);
+	}
+
+	// All other `JUMPDEST`'s that were not followed by a tag are treated as a fatal error.
+	for (auto const& item: m_items)
+		if (item.type() == AssemblyItemType::Operation && item.instruction() == Instruction::JUMPDEST)
+			solThrow(AssemblyImportException, "JUMPDEST instruction found that was not followed by tag.");
```

Comment:
Why not simplify this...

```suggestion
	for (auto current = begin(_code); current != end(_code); ++current)
		if (m_items.emplace_back(createAssemblyItemFromJSON(*current, m_sourceList)).type() == AssemblyItemType::Tag)
			solRequire(
				++current != end(_code) &&
				createAssemblyItemFromJSON(*current, m_sourceList)  == Instruction::JUMPDEST,
				AssemblyImportException,
				"JUMPDEST expected after tag."
			);
```
Maybe a local variable or two would help reading this, but this should do.